flepimop/gempyor_pkg/tests/parameters/test_parameters_class.py

```diff
@@ -0,0 +1,508 @@
+            )
+            assert params.pdata["gamma"]["stacked_modifier_method"] == "sum"
+            assert set(params.pdata["Ro"].keys()) == {
+                "idx",
+                "dist",
+                "stacked_modifier_method",
+            }
+            assert params.pdata["Ro"]["idx"] == 2
+            assert isinstance(params.pdata["Ro"]["dist"], partial)
+            assert partials_are_similar(
+                params.pdata["Ro"]["dist"], partial(np.random.uniform, 1.0, 2.0)
+            )
+            assert params.pdata["Ro"]["stacked_modifier_method"] == "product"
+
+            # The `pnames` attribute
+            assert params.pnames == ["sigma", "gamma", "Ro"]
+
+            # The `pnames2pindex` attribute
+            assert params.pnames2pindex == {"sigma": 0, "gamma": 1, "Ro": 2}
+
+            # The `stacked_modifier_method` attribute
+            assert params.stacked_modifier_method == {
+                "sum": ["gamma"],
+                "product": ["sigma", "ro"],
+                "reduction_product": [],
+            }
+
+    def test_picklable_lamda_alpha(self) -> None:
+        # Setup
+        simple_parameters = create_confuse_subview_from_dict(
+            "parameters", {"sigma": {"value": 0.1}}
+        )
+        params = Parameters(
+            simple_parameters,
+            ti=date(2024, 1, 1),
+            tf=date(2024, 1, 10),
+            subpop_names=["1", "2"],
+        )
+
+        # Attribute error if `alpha_val` is not set
+        with pytest.raises(AttributeError):
+            params.picklable_lamda_alpha()
+
+        # We get the expected value when `alpha_val` is set
+        params.alpha_val = None
+        assert params.picklable_lamda_alpha() == None
+
+    def test_picklable_lamda_sigma(self) -> None:
+        # Setup
+        simple_parameters = create_confuse_subview_from_dict(
+            "parameters", {"sigma": {"value": 0.1}}
+        )
+        params = Parameters(
+            simple_parameters,
+            ti=date(2024, 1, 1),
+            tf=date(2024, 1, 10),
+            subpop_names=["1", "2"],
+        )
+
+        # Attribute error if `sigma_val` is not set
+        with pytest.raises(AttributeError):
+            params.picklable_lamda_sigma()
+
+        # We get the expected value when `sigma_val` is set
+        params.sigma_val = None
+        assert params.picklable_lamda_sigma() == None
+
+    def test_get_pnames2pindex(self) -> None:
+        simple_parameters = create_confuse_subview_from_dict(
+            "parameters",
+            {"sigma": {"value": 0.1}, "gamma": {"value": 0.2}, "eta": {"value": 0.3}},
+        )
+        params = Parameters(
+            simple_parameters,
+            ti=date(2024, 1, 1),
+            tf=date(2024, 1, 10),
+            subpop_names=["1", "2"],
+        )
+        assert params.get_pnames2pindex() == params.pnames2pindex
+        assert params.pnames2pindex == {"sigma": 0, "gamma": 1, "eta": 2}
```

Comment:
is there a method contract / requirement that they should have this indexing order?

Reply:
Seems worth delving on this a bit to whether it is assumed, but I'm wary (from a loose-coupling perspective) of providing / testing unnecessary guarantees.

Reply:
This constraint has been loosened in the unit tests now.